src/Razor/Microsoft.AspNetCore.Razor.Language/test/IntegrationTests/ComponentDiagnosticRazorIntegrationTest.cs

```diff
@@ -147,24 +146,35 @@ public void Component_StartsWithLowerCase_ReportsError()
                 diagnostic.GetMessage(CultureInfo.CurrentCulture));
         }
 
-        [Fact]
-        public void Element_DoesNotStartWithLowerCase_ReportsWarning()
+        [Theory]
+        [InlineData("PossibleComponent", true, true)]
+        [InlineData("PossibleComponent", false, true)]
+        [InlineData("繁体字", true, true)]
+        [InlineData("繁体字", false, false)]
+        public void Component_NotFound_ReportsWarning(string componentName, bool supportLocalizedComponentNames, bool diagnosticExpected)
         {
             // Arrange & Act
-            var generated = CompileToCSharp(@"
-<PossibleComponent></PossibleComponent>
+            var generated = CompileToCSharp(@$"
+<{componentName}></{componentName}>
 
-@functions {
-    public string Text { get; set; } = ""text"";
-}");
+@functions {{
+    public string Text {{ get; set; }} = ""text"";
+}}", supportLocalizedComponentNames: supportLocalizedComponentNames);
 
             // Assert
-            var diagnostic = Assert.Single(generated.Diagnostics);
-            Assert.Equal("RZ10012", diagnostic.Id);
-            Assert.Equal(RazorDiagnosticSeverity.Warning, diagnostic.Severity);
-            Assert.Equal(
-                "Found markup element with unexpected name 'PossibleComponent'. If this is intended to be a component, add a @using directive for its namespace.",
-                diagnostic.GetMessage(CultureInfo.CurrentCulture));
+            if (diagnosticExpected)
```

Comment:
Could we make this a separate test? it feels awkward to have a branch in the tests.